File: features/user/topchat/src/main/java/com/tokopedia/topchat/chattemplate/data/repository/EditTemplateRepositoryImpl.java
package com.tokopedia.topchat.chattemplate.data.repository;

import com.tokopedia.topchat.chattemplate.data.factory.EditTemplateChatFactory;
import com.tokopedia.topchat.chattemplate.view.viewmodel.EditTemplateUiModel;

import java.util.HashMap;

import rx.Observable;

/**
 * Created by stevenfredian on 11/27/17.
 */

public class EditTemplateRepositoryImpl implements EditTemplateRepository {

    private EditTemplateChatFactory templateChatFactory;

    public EditTemplateRepositoryImpl(EditTemplateChatFactory templateChatFactory) {
        this.templateChatFactory = templateChatFactory;
    }

    @Override
    public Observable<EditTemplateUiModel> editTemplate(int index, HashMap<String, Object>
            parameters, boolean isSeller) {
        return templateChatFactory.createCloudEditTemplateDataSource().editTemplate(index, parameters, isSeller);
    }

    @Override
    public Observable<EditTemplateUiModel> createTemplate(HashMap<String, Object> parameters) {
        return templateChatFactory.createCloudEditTemplateDataSource().createTemplate(parameters);
    }

    @Override
    public Observable<EditTemplateUiModel> deleteTemplate(int index, boolean isSeller) {
        return templateChatFactory.createCloudEditTemplateDataSource().deleteTemplate(index, isSeller);
    }
}

File: features/user/topchat/src/main/java/com/tokopedia/topchat/chattemplate/data/repository/EditTemplateRepositoryImpl.kt
package com.tokopedia.topchat.chattemplate.data.repository

import com.tokopedia.topchat.chattemplate.data.source.CloudEditTemplateChatDataSource
import com.tokopedia.topchat.chattemplate.domain.pojo.TemplateData
import javax.inject.Inject

class EditTemplateRepositoryImpl @Inject constructor(
    private val cloudEditTemplateChatDataSource: CloudEditTemplateChatDataSource
): EditTemplateRepository {

    override suspend fun editTemplate(
        index: Int,
        map: Map<String, Any>,
        isSeller: Boolean
    ): TemplateData {
        return cloudEditTemplateChatDataSource.editTemplate(index, map, isSeller)
    }

    override suspend fun createTemplate(map: Map<String, Any>): TemplateData {
        return cloudEditTemplateChatDataSource.createTemplate(map)
    }

    override suspend fun deleteTemplate(index: Int, isSeller: Boolean): TemplateData {
        return cloudEditTemplateChatDataSource.deleteTemplate(index, isSeller)
    }
}